File: Source/OldCode/Archive/2004 TVA Code Library/Source/EE/Phasor/IDataFrame.vb
'***********************************************************************
'  IDataFrame.vb - Data frame interface
'  Copyright © 2004 - TVA, all rights reserved
'
'  Build Environment: VB.NET, Visual Studio 2003
'  Primary Developer: James R Carroll, System Analyst [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  ---------------------------------------------------------------------
'  01/14/2005 - James R Carroll
'       Initial version of source generated
'
'***********************************************************************

Namespace EE.Phasor

    ' This interface represents the protocol independent representation of any frame of phasor data.
    Public Interface IDataFrame

        Inherits IChannelFrame

        ReadOnly Property DataCells() As DataCellCollection

    End Interface

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IDataFrame.cs
//*******************************************************************************************************
//  IDataFrame.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of any kind of data frame that contains
    /// a collection of <see cref="IDataCell"/> objects.
    /// </summary>
    public interface IDataFrame : IChannelFrame
    {
        /// <summary>
        /// Gets or sets <see cref="IConfigurationFrame"/> associated with this <see cref="IDataFrame"/>.
        /// </summary>
        IConfigurationFrame ConfigurationFrame { get; set; }

        /// <summary>
        /// Gets reference to the <see cref="DataCellCollection"/> for this <see cref="IDataFrame"/>.
        /// </summary>
        new DataCellCollection Cells { get; }

        /// <summary>
        /// Gets or sets the parsing state for the this <see cref="IDataFrame"/>.
        /// </summary>
        new IDataFrameParsingState State { get; set; }
    }
}